packages/next/src/utilities/initPage/handleAuthRedirect.ts

```diff
@@ -21,11 +21,11 @@ export const handleAuthRedirect = ({ config, route, searchParams, user }: Args):
     delete searchParams.redirect
   }
 
-  const redirectRoute = encodeURIComponent(
-    route + Object.keys(searchParams ?? {}).length
+  const redirectRoute =
+    (route !== '/admin' ? route : '') +
```

Comment:
This needs to be pulled from the config.

Reply:
Hi Dan, just clarifying as I'm also keen for this fix... in the diff, line 25: route !== '/admin' did you mean that the route needs to come from config i.e. because we aren't all using /admin? Cheers, Nat.